src/idl_gen_general.cpp

```diff
@@ -409,6 +409,23 @@ static std::string GenDefaultValue(const LanguageParameters &lang, const Value &
            : value.constant;
 }
 
+static std::string GenEnumDefaultValue(const LanguageParameters &lang, const FieldDef &field) {
+  if (lang.language == GeneratorOptions::kCSharp) {
+    auto enum_def = field.value.type.enum_def;
+    for (auto it = enum_def->vals.vec.begin();
+    it != enum_def->vals.vec.end(); ++it) {
+      auto enum_val = **it;
+      if (enum_val.value == StringToInt(field.value.constant.c_str())) {
+        return enum_val.name;
+      }
+    }
+  }
+
+  // fail through
+  return field.value.constant;
```

Comment:
This is not a good default, as this will generate code like "Color.2" which will give weird errors. Maybe have this function also generate the "Color." part, so the default doesn't need to have it.